assets/state-container-toolkit/0500_daemonset.yaml

```diff
@@ -78,13 +78,17 @@ spec:
             readOnly: true
             mountPath: /bin/entrypoint.sh
             subPath: entrypoint.sh
-          - name: nvidia-run-path
-            mountPath: /run/nvidia
-            mountPropagation: Bidirectional
+          - name: toolkit-pid-file
+            mountPath: /run/nvidia/toolkit.pid
```

Comment:
These are good questions. With my limited testing, the toolkit container appears to acquire the exclusive lock just fine, even after restarting the pod or uninstalling / re-installing the operator. But definitely something I can look into more.

Note, I decided to remove the `/run/nvidia` mount to avoid mounting the driver install directory twice -- once at `/driver-root` in the container and again at `/run/nvidia/driver`. If preferred, I can reintroduce the original `/run/nvidia` mount, or we can update the toolkit container to create the pid file at another path, like `/run/nvidia/toolkit/toolkit.pid`.

Reply:
Thanks @elezar. I prefer to create the pid file at `/run/nvidia/toolkit/toolkit.pid` instead, so let's go ahead with the change you are making to the toolkit container.

Reply:
I have updated this now. We now mount `/run/nvidia/toolkit` as `DirectoryOrCreate` in the toolkit container. We also set the `TOOLKIT_PID_FILE=/run/nvidia/toolkit/toolkit.pid` envvar explicitly.